### PR TITLE
[CDAP-5338] [CDAP-5294] Feature/ui tracker navigator config

### PR DESCRIPTION
**Pull request by elfenheart**
- Add optional settings to cloudera navigator config
- Add ability to edit configuration

### CODE REVIEW

File: cdap-ui/app/features/tracker/controllers/integrations-ctrl.js [hunk 1]
```diff
@@ -63,16 +83,28 @@ class TrackerIntegrationsController {
 
     this.pollId = null;
 
+    $scope.$watch(() => {
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/integrations-ctrl.js]:
```
$scope.$watch('TrackerIntegrationsController.navigatorSetup.isOpen', () => {...});
```
Is it possible to use just the controller variable instead of function callback?

File: cdap-ui/app/features/tracker/controllers/integrations-ctrl.js [hunk 2]
```diff
@@ -29,15 +29,35 @@ class TrackerIntegrationsController {
...
+      navigator: false,
+      kafka: false
+    };
+
     this.navigatorInfo = {
-      brokerString: '',
-      hostname: '',
-      username: '',
-      password: ''
+      navigatorConfig: {
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/integrations-ctrl.js]: Should we get this from a configuration factory just like we do in dag directive. It might be easier to plugin it in another factory in case if we need to integrate other management tools. Just a suggestion, if the changes are too big then we can move it to another PR. We name this file as `integrations-ctrl.js` but the implementation is too specific for navigator.

File: cdap-ui/app/features/tracker/controllers/integrations-ctrl.js [hunk 3]
```diff
@@ -63,16 +83,28 @@ class TrackerIntegrationsController {
...
-          return config.name === 'metadata.updates.kafka.broker.list';
+        let zookeeperQuorum = res.filter( (config) => {
+          return config.name === 'zookeeper.quorum';
+        });
+
+        let zookeeperKafka = res.filter( (config) => {
+          return config.name === 'kafka.zookeeper.namespace';
         });
-        this.navigatorInfo.brokerString = filtered[0].value;
+        this.navigatorInfo.metadataKafkaConfig.zookeeperString = zookeeperQuorum[0].value + '/' + zookeeperKafka[0].value;
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/integrations-ctrl.js]: Could there be a possibility that one of the values `zookeeperQuorum or zookeeperKafka` is an empty array?

Comment (elfenheart) [inline cdap-ui/app/features/tracker/controllers/integrations-ctrl.js]: No, if this value is not set, CDAP won't start

**Comment by ajainarayanan** [comment]
One minor bug: 
- Setup the app.
- view config and edit it. 
- after changing something without saving hit cancel.

The edited value still remains in the view. We should discard the changes if it is not saved.

Otherwise :+1: LGTM
